src/app/auth/auth.guard.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, CanDeactivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate, CanDeactivate<unknown> {
+
+  constructor(
+    private router: Router
+  ){}
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+      if(localStorage.getItem('token') || sessionStorage.getItem('token')){
+        return true;
+      }
+      this.router.navigate(['login']);
```

Comment:
Optionally. It would be good to put all navigation into separate navigation service